src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,97 @@
-import { useState } from 'react';
+import React from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+const InitialFieldsOfMovie: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+type Field =
+  'title'
+  | 'description'
+  | 'imgUrl'
+  | 'imdbUrl'
+  | 'imdbId';
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = React.useState(0);
+  const [fieldsOfMovie, setFieldsOfMovie] = React
+    .useState(InitialFieldsOfMovie);
+
+  const isDisabled = !(
+    fieldsOfMovie.title
+    && fieldsOfMovie.imgUrl
+    && fieldsOfMovie.imdbUrl
+    && fieldsOfMovie.imdbId);
+
+  const submitHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    onAdd({
+      title: fieldsOfMovie.title,
+      description: fieldsOfMovie.description,
+      imgUrl: fieldsOfMovie.imgUrl,
+      imdbUrl: fieldsOfMovie.imdbUrl,
+      imdbId: fieldsOfMovie.imdbId,
+    });
+
+    setCount((prevValue) => prevValue + 1);
+    setFieldsOfMovie(InitialFieldsOfMovie);
+  };
+
+  const changeFieldHandler = (name: Field, value: string) => {
+    setFieldsOfMovie(prevFields => ({ ...prevFields, [name]: value }));
+  };
```

Comment:
```suggestion
  const changeFieldHandler = (key: Field) => ( value: string) => {
    setFieldsOfMovie(prevFields => ({ ...prevFields, [key]: value }));
  };
```
You can make it closure for best view )